agent/client/client.go

```diff
@@ -102,22 +106,24 @@ func New(cfg configGetter, supervisor supervisor, r *runner.Runner, connectionCh
 		cus:               cus,
 		logStore:          logStore,
 	}
+
+	var err error
+	if out.cache, err = cache.New(cfg.Get().Cache); err != nil {
+		out.l.Infof("Failed to init cache: %s. Initializing cachelless client.", err)
+		out.cache = &cache.Dummy{}
```

Comment:
I think we should fail instead of fallback to cache-less mode.

Reply:
Not sure about that, let's discuss when you come back from vacation